app/components/SCForm.js

```diff
@@ -94,28 +94,56 @@ class SCForm extends Component {
   }
 
   onChange(value) {
-    this.setState({ value });
-  }
+    const formInfo = this.props.navigation.state.params.form;
+    const length = formInfo.fields.length;
+    let newState = {};
+    let fieldValue, fieldKey, max, min;
+    let fieldValidations = [];
 
+    for (let i = 0; i < length; i++) {
+      // input
+      let type = formInfo.fields[i].type;
+      let field = formInfo.fields[i];
+      min = formInfo.fields[i].minimum;
+      max = formInfo.fields[i].maximum;
+      fieldKey = formInfo.fields[i].field_key;
+      fieldValue = value[formInfo.fields[i].field_key];
+
+      if (fieldValue !== undefined) {
+        if (type === 'number') {
+          fieldValue = +[fieldValue];
+          ruleRunner(field.fieldKey, field.fieldLabel,
```

Comment:
this needs to be assigned to a variable, which then needs to pushed onto `fieldValidations`